mlflow_utils.py

```diff
@@ -0,0 +1,97 @@
+import keras
+import mlflow
+from keras.models import Model
+from mlflow import pyfunc
+import temporal_fusion_transformers as tft
+
+
+class KerasModelWrapper(pyfunc.PythonModel):
+    def __init__(self, model: Model):
+        self.model = model
+
+    def load_context(self, context):
+        """Loads the model from the artifact path"""
+        self.model = keras.models.load_model(context.artifacts["model"])
+        self._hack_build()
+        self.model.load_weights(context.artifacts["weights"])
+
+    def predict(self, context, model_input):
+        return self.model.predict(model_input)
+
+    def _hack_build(self):
+        x, _ = tft.prepare_data_samples(
+            n_samples=1,
+            df_daily_input=tft.df_input_scl,
+            df_target=tft.df_target_1m_pct,
+            sampled_day="2000-01-01",
+            min_context=365,
+            context_length=365,
+            country="US",
+        )
+        self.model(x)
+
+
+def log_keras_model(model: Model, artifact_path: str):
+    """Log Keras model to MLflow as an artifact with custom serialization.
+
+    Args:
+        model (Model): The model to log.
+        artifact_path (str): The artifact path in MLflow.
+    """
+    model.save_weights("tft.weights.h5")
```

Comment:
Shouldn't the filenames for the weights and model be dynamic instead of hard-coded?

Reply:
For example, in the `load_keras_model` function the uri is an argument.